perception/navigator_vision/nodes/pinger_finder.py

```diff
@@ -0,0 +1,170 @@
+            print "delta enu frame: {}".format(delta)
+            p1 = p0 + point_distance * delta / npl.norm(delta)
+            line_coeffs = np.array([[p0[0], p0[1], p1[0], p1[1]]]) # p0 and p1 define a line
+            visualize_line(Point(p0[0], p0[1], 0), Point(p1[0], p1[1], 0))
+            if listen:
+               line_array = np.append(line_array, line_coeffs, 0)
+               sample_amplitudes = np.append(sample_amplitudes, processed_ping.amplitude)
+            else:
+                print "Not currently listening"
+        except (tf.LookupException, tf.ConnectivityException, tf.ExtrapolationException) as e:
+            print e
+        # delete softest samples if we have over max_samples
+        if len(line_array) >= max_samples:
+            softest_idx = np.argmin(sample_amplitudes)
+            line_array = np.delete(line_array, softest_idx, axis=0)  
+            sample_amplitudes = np.delete(sample_amplitudes, softest_idx)
+        print "Number of good samples: {}".format(line_array.shape)
+    else:
+        print "Untrustworthy pinger heading. Freq = {} kHZ".format(processed_ping.freq)
+
+def visualize_pinger(event):
+    marker = Marker()
+    marker.ns = "pinger"
+    marker.header.stamp = rospy.Time(0)
+    marker.header.frame_id = "/enu"
+    marker.type = marker.CUBE
+    marker.action = marker.ADD
+    marker.scale.x = 1.0
+    marker.scale.y = 1.0
+    marker.scale.z = 1.0
+    marker.color.g = 1.0
+    marker.color.a = 1.0
+    marker.pose.orientation.w = 1.0
+    marker.pose.position = pinger_position
+    if pinger_position != Point(0, 0, 0):
+        print "vis pinger"
+        viz_pub.publish(marker)
+    else:
+        print "pinger position: {}".format(pinger_position)
+
+def visualize_line(p0, p1):
+    global debug_arrow_id
+    marker = Marker()
+    marker.ns = "pinger/debug"
+    marker.header.stamp = rospy.Time(0)
+    marker.header.frame_id = "/enu"
+    marker.id = debug_arrow_id
+    debug_arrow_id += 1
+    marker.type = marker.ARROW
+    marker.action = marker.ADD
+    marker.lifetime = rospy.Duration(10, 0)
+    marker.points.append(p0)
+    marker.points.append(p1)
+    marker.scale.x = 0.5
+    marker.scale.y = 1.0
+    marker.scale.z = 1.0
+    marker.color.g = 1.0
+    marker.color.b = 1.0
+    marker.color.a = 1.0
+    print "viz line"
+    viz_pub.publish(marker)
+    
+def set_listen(listen_status):
+    global listen
+    listen = listen_status.data
+    if listen:
+        print "PINGERFINDER: setting listening to on"
+    else:
+        print "PINGERFINDER: setting listening to off"
+    return {'success': True, 'message': ""}
+
+def set_freq(msg):
+    global target_freq, line_array, sample_amplitudes, pinger_position
+    target_freq = msg.frequency
+    line_array = np.empty((0, 4), float)
+    sample_amplitudes = np.empty((0, 0), float)
+    pinger_position = Point(0, 0, 0)
+    return SetFrequencyResponse()
+
+
+rospy.init_node('pinger_finder')
```

Comment:
I agree, also all the stuff at the top should be enclosed too, especially that publisher (which probably should go after the init)